bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/components/UIComponentSitemapProvider.java

```diff
@@ -412,50 +427,78 @@ private void addIconColor(EList<ColorArray> iconColor, UIComponent component) {
+                        List<String> conditionsString = getRuleConditions(sourceIcon.toString(), argument);
                         IconRuleImpl iconRule = (IconRuleImpl) SitemapFactory.eINSTANCE.createIconRule();
-                        ConditionImpl condition = (ConditionImpl) SitemapFactory.eINSTANCE.createCondition();
-                        condition.setItem(matcher.group("item"));
-                        condition.setCondition(matcher.group("condition"));
-                        condition.setSign(matcher.group("sign"));
-                        condition.setState(matcher.group("state"));
-                        iconRule.eSet(SitemapPackage.ICON_RULE__CONDITIONS, condition);
-                        iconRule.setArg(matcher.group("arg"));
-                        iconDef.add(iconRule);
-                    } else {
-                        logger.warn("Syntax error in icon rule '{}' for widget {}", sourceIcon, component.getType());
+                        iconRule.eSet(SitemapPackage.ICON_RULE__ARG, argument);
+                        List<ConditionImpl> conditions = getConditions(conditionsString, component, "iconrules");
+                        iconRule.eSet(SitemapPackage.ICON_RULE__CONDITIONS, conditions);
+                        icon.add(iconRule);
                     }
                 }
             }
         }
     }
 
+    private List<ConditionImpl> getConditions(List<String> conditionsString, UIComponent component, String key) {
+        List<ConditionImpl> conditions = new ArrayList<>();
+        for (String conditionString : conditionsString) {
+            Matcher matcher = CONDITION_PATTERN.matcher(conditionString);
+            if (matcher.matches()) {
+                ConditionImpl condition = (ConditionImpl) SitemapFactory.eINSTANCE.createCondition();
+                condition.setItem(matcher.group("item"));
+                condition.setCondition(matcher.group("condition"));
+                condition.setSign(matcher.group("sign"));
+                condition.setState(matcher.group("state"));
+                conditions.add(condition);
+            } else {
+                logger.warn("Syntax error in {} rule condition '{}' for widget {}", key, conditionString,
+                        component.getType());
+            }
+        }
+        return conditions;
+    }
+
+    private String getRuleArgument(String rule) {
+        int argIndex = rule.lastIndexOf("=") + 1;
+        return rule.substring(argIndex).trim();
+    }
+
+    private List<String> getRuleConditions(String rule, @Nullable String argument) {
+        String conditions = rule;
+        if (argument != null) {
+            conditions = rule.substring(0, rule.lastIndexOf(argument)).trim();
+            if (conditions.endsWith("=")) {
+                conditions = conditions.substring(0, conditions.length() - 1);
+            }
+        }
+        List<String> conditionsList = List.of(conditions.split(" AND "));
```

Comment:
Does UI systematically adds space around the AND ?

Reply:
Yes, it does.